arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -2021,24 +1876,30 @@ Result RocksDBCollection::lookupDocumentVPack(
       RocksDBColumnFamilyManager::Family::Documents);
   rocksdb::Status s =
       snapshot
-          ? mthd->SingleGet(snapshot->getSnapshot(), *family, key->string(), ps)
-          : mthd->Get(family, key->string(), &ps, readOwnWrites);
+          ? mthd->SingleGet(
+                static_cast<RocksDBEngine::RocksDBSnapshot const*>(snapshot)
+                    ->getSnapshot(),
+                *family, key->string(), ps)
+          : mthd->Get(family, key->string(), &ps,
+                      static_cast<ReadOwnWrites>(options.readOwnWrites));
 
   if (!s.ok()) {
     return rocksutils::convertStatus(s);
   }
 
   TRI_ASSERT(ps.size() > 0);
-  cb(documentId, VPackSlice(reinterpret_cast<uint8_t const*>(ps.data())));
-
-  if (withCache && cache != nullptr) {
+  if (options.fillCache && cache != nullptr) {
     // write entry back to cache
     cache::Cache::SimpleInserter<DocumentCacheType>{
         static_cast<DocumentCacheType&>(*cache), key->string().data(),
         static_cast<uint32_t>(key->string().size()), ps.data(),
         static_cast<uint64_t>(ps.size())};
   }
 
+  auto data = buffer.release();
+  cb(token, std::move(data));
+  buffer.acquire(std::move(data));
```

Comment:
`data` is moved twice here. While this may work (if `cb` does not steal the value from `data`), it looks strange to the reader of the code.

Reply:
Changed